### DISCUSSION TITLE
Is there any way to set a field as read only in orchard

### DISCUSSION THREAD

**Discussion by AmadeuAntunes** [OrchardCMS/OrchardCore#8461] [Q&A]
I'm trying to create a textbox field as readonly, is it possible using some setting or "recipe"?

![image](https://user-images.githubusercontent.com/47931101/106294384-03bf4980-6247-11eb-8706-d9c910d41d7e.png)

**Comment by tropcicstefan** [comment]
There is no readonly text field editor "out of the box" in Orchard. But it should be pretty simple. First look at this [link](https://docs.orchardcore.net/en/dev/docs/reference/modules/ContentFields/#creating-custom-editors).

In essence you should make new textfield editor.

Look at OrchardCore.ContentFields-> Views. There focus on TextField.Edit.cshtml and TextField.Option.cshtml. Copy those two views to your module and change their names to TextField-Readonly.Edit.cshtml and TextField-Readonly.Option.cshtml. In Edit view add `readonly` HTML attribute to input field. In option view change option line like so
 `<option value="Readonly" selected="@(currentEditor == "Readonly")">@T["Readonly"]</option>`

On your content part change editor of text field.

**Reply by AmadeuAntunes** [reply]
I've placed it in **Views** folder and worked, is it possible to move it for other folder inside **Views** ?

**Comment by panayiotisp** [comment]
You can add them in a separate module if you want to have them in a different area than your other module's views

**Comment by hishamco** [comment]
I'm wondering how you can add the values for such fields.

**Comment by panayiotisp** [comment]
This will be done from code (C#) or from the front end part itself.

**Reply by hishamco** [reply]
It could be, for that I prefer @tropcicstefan suggestion https://github.com/OrchardCMS/OrchardCore/discussions/8461#discussioncomment-321269

**Reply by panayiotisp** [reply]
I am not contradicting @tropcicstefan. I am actually using his suggestion as well. I just replied to the part you asked how would the field values be inserted/updated if the field in the admin is readonly. In this case you can update them either from the frontend of the website or from code (like for example a background task that will sync data between another system and your website content item store).

**Reply by hishamco** [reply]
It's a good addition that can help some of us. I might add it to Orchard Core Contrib if there's no plan to have it in OC
